src/mainchain/MainchainGatewayV3.sol

```diff
@@ -461,8 +465,9 @@ contract MainchainGatewayV3 is
   /**
    * @inheritdoc GatewayV3
    */
-  function _getTotalWeight() internal view override returns (uint256) {
-    return _totalOperatorWeight;
+  function _getTotalWeight() internal view override returns (uint256 totalWeight) {
+    totalWeight = _totalOperatorWeight;
+    if (totalWeight == 0) revert ErrNullTotalWeightProvided(msg.sig);
```

Comment:
should we check minimumVoteWeight as well?

Reply:
updated.